### ISSUE TITLE
Question: feature detection & polyfilling on the client side, possible with babel-loader?

### ISSUE THREAD

**Issue by rondonjon** [babel/babel-loader#843]
It is a common practice now to run builds several times, at least once for "older" browsers and again for "modern" browsers, with different sets of polyfills. This is IMO not the right way or the right stage in the build/load process to hande feature detection and polyfilling. IMO it should happen entirely at runtime.

[There seems to be no easy solution to do this with the common babel / babel-loader setup.](//https://stackoverflow.com/questions/58261075/how-to-detect-features-and-lazy-load-only-the-needed-polyfills-with-babel-core)

The reason is probably that the babel-loader generates code which "hard-wires" all polyfills with `require()`.

Wouldn't it be possible to wrap these in feature tests and loading them only conditionally, with a mechanism like dynamic requires? (I know, these might night to be polyfilled themselves and hence be the only exception to the general approach.)

I've looked at the `customize` option of the babel-loader but found no documentation on the actual customization options, and the behaviour that was implemented in the [webpack example](https://webpack.js.org/loaders/babel-loader/#customized-loader) seems to receive ready-made code from babel inside `result.code`, whereas the `result.ast` (which might be easier to extend with the desired behavour) is `null`.

So, to sum this up, any suggestions on how to make feature detection and polyfills a client-side, download-only-what-you-need solution with babel & babel-loader?
